webpack/webpack5-localization-plugin/src/LocalizationPlugin.ts

```diff
@@ -97,13 +122,24 @@ export class LocalizationPlugin implements WebpackPluginInstance {
     string,
     Map<string, ILocaleFileObject | string | ReadonlyMap<string, string>>
   > = new Map();
-  private readonly _stringPlaceholderMap: Map<string, IStringPlaceholder> = new Map();
+  private readonly _stringPlaceholderBySuffix: Map<string, IStringPlaceholder> = new Map();
+  private readonly _customDataPlaceholderBySuffix: Map<string, ICustomDataPlaceholder> = new Map();
+  private readonly _customDataPlaceholderByCustomValueFunction: WeakMap<
+    ValueForLocaleFn,
+    ICustomDataPlaceholder
+  > = new WeakMap();
   private _passthroughLocaleName!: string;
   private _defaultLocale!: string;
   private _noStringsLocaleName!: string;
   private _fillMissingTranslationStrings!: boolean;
-  private _formatLocaleForFilename!: (loc: string) => string;
+  /**
+   * @remarks
+   * Include the `chunk` parameter so that the functions arity is the same as the
+   * `ValueForLocaleFn` type.
+   */
+  private _formatLocaleForFilename!: (loc: string, chunk: unknown) => string;
   private readonly _pseudolocalizers: Map<string, (str: string) => string> = new Map();
+  private _customValueSuffixCounter: number = 0;
```

Comment:
Using a counter makes the algorithm nondeterministic and therefore not play nicely with running in watch mode. That's why I swapped the string replacement engine to use hashes instead of an auto-incrementing id.

Reply:
I added a unique ID parameter, but I can't say I love that.